tests/behaviour/cli.feature

```diff
@@ -7,4 +7,64 @@ Feature: Test the csvcubed Command Line Interface.
+    And the file at "out/title-of-the-code-list.table.json" should exist
+    And the file at "out/validation-errors.json" should exist
+    And the validation-errors.json file should contain
+    """
+    "offending_value": "http://purl.org/dc/aboutdcmi#DCMI"
+    """
+    Then remove test log files
+
+  Scenario: The csvcubed code-list build command should display the logging in accordance with the log level set at debug.
+    Given the existing test-case file "readers/code-list-config/v1.0/code_list_config_produces_error.json"
+    When the csvcubed CLI is run with "code-list build readers/code-list-config/v1.0/code_list_config_produces_error.json --log-level debug"
+    Then the command line output should display the log message
+    """
+    csvcubed.utils.cli - WARNING - Schema Validation Error: $.creator - 'http://purl.org/dc/aboutdcmi#DCMI' is not one of ['http://dbpedia.org/resource/Open_Knowledge_Foundation', 'http://statistics.data.gov.uk', 'https:/
+    """
+    And the command line output should display the log message
+    """
+    csvcubed.utils.cli - ERROR - Validation Error: An error was encountered when validating the cube. The error occurred in '["('concepts', 3)", 'existing_concept_uri']' and was reported as ''This will produce an error' does not look like a URI.'
+    """
+    And the command line output should display the log message
+    """
+    csvcubed.readers.codelistconfig.codelist_schema_versions - INFO - Using schema version 1.0
+    """
+    And the command line output should display the log message
+    """
+    csvcubed.utils.json - DEBUG - Loading JSON from URL https://purl.org/csv-cubed/code-list-config/v1.0
+    """
+
+  Scenario: The csvcubed code-list build command should display the logging in accordance with the log level set at critical.
+    Given the existing test-case file "readers/code-list-config/v1.0/code_list_config_produces_critical_error.json"
+    When the csvcubed CLI is run with "code-list build readers/code-list-config/v1.0/code_list_config_produces_critical_error.json --log-level debug"
+    Then the command line output should display the log message
+    """
+    csvcubed.cli.entrypoint - CRITICAL - Traceback (most recent call last):
+    """
+
+  Scenario: The csvcubed code-list build command will continue when there is a validation error and build a code list csvw in a given out directory.
+    Given the existing test-case file "readers/code-list-config/v1.0/code_list_config_produces_error.json"
+    When the csvcubed CLI is run with "code-list build readers/code-list-config/v1.0/code_list_config_produces_error.json --ignore-validation-errors --out testout"
+    Then the file at "testout/title-of-the-code-list.csv" should exist
+    And the file at "testout/title-of-the-code-list.csv-metadata.json" should exist
+    And the file at "testout/title-of-the-code-list.table.json" should exist
+    And the command line output should display the log message
```

Comment:
Can you test that there is indeed a validation error/that the log says that it's continuing even though it found validation errors?

Could you also add a step which asserts that the status_code of executed command is zero (i.e. the app says it succeeded). You should do this for every scenario where the app is supposed to succeed. The `the csvcubed CLI should succeed` should be sufficient to test this (as well as the fact that it says `build complete` in the log).

Can you also make sure the ones which should fail use the `the csvcubed CLI should fail with status code {status_code}` step to ensure the app tells the console that it failed.